app/src/main/java/com/anzelmas/quizzando/QuizQuestionActivity.kt

```diff
@@ -14,6 +15,145 @@ import androidx.databinding.DataBindingUtil
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_quiz_question)
+
+        userName = intent.getStringExtra(Constants.USER_NAME)
+
+        questionList = Constants.getQuestions()
+//        Log.i("Question size", "${questionList!!.size}")
+
+        putQuestion()
+
+        optionOne_TextView.setOnClickListener(this)
+        optionTwo_textView.setOnClickListener(this)
+        optionThree_TextView.setOnClickListener(this)
+        optionFour_TextView.setOnClickListener(this)
+        submit_Button.setOnClickListener(this)
+    }
+
+    private fun putQuestion() {
+        val question = questionList!![currentPosition - 1]
+
+        defaultOptions()
+        submit_Button.text = "SUBMIT"
+
+        progressBar.progress = currentPosition
+        progressBar_textView.text = "$currentPosition" + "/" + progressBar.max
+
+
+        question_TextView.text = question.question
+        question_image_imageView.setImageResource(question.image)
+        optionOne_TextView.text = question.optionOne
+        optionTwo_textView.text = question.optionTwo
+        optionThree_TextView.text = question.optionThree
+        optionFour_TextView.text = question.optionFour
+    }
+
+    private fun defaultOptions() {
+        val options = ArrayList<TextView>()
+        options.add(0, optionOne_TextView)
+        options.add(1, optionTwo_textView)
+        options.add(2, optionThree_TextView)
+        options.add(3, optionFour_TextView)
+
+        for (option in options) {
+            option.setTextColor(Color.parseColor("#313030"))
+            option.typeface = Typeface.DEFAULT
+            option.background = ContextCompat.getDrawable(
+                this, R.drawable.default_option_border_bg
+            )
+        }
+    }
+
+    private fun selectedOptionView(textView: TextView, selectedOptionNumber: Int) {
+        defaultOptions()
+
+        selectedOption = selectedOptionNumber
+
+        textView.setTextColor(Color.parseColor("#313030"))
+        textView.setTypeface(textView.typeface, Typeface.BOLD)
+        textView.background = ContextCompat.getDrawable(this, R.drawable.selected_option_border_bg)
+    }
+
+    override fun onClick(view: View?) {
+
+        when (view?.id) {
+            R.id.optionOne_TextView -> selectedOptionView(optionOne_TextView, 1)
+            R.id.optionTwo_textView -> selectedOptionView(optionTwo_textView, 2)
+            R.id.optionThree_TextView -> selectedOptionView(optionThree_TextView, 3)
+            R.id.optionFour_TextView -> selectedOptionView(optionFour_TextView, 4)
+            R.id.submit_Button -> {
+                if (selectedOption == 0) {
+                    currentPosition++
+                    when {
+                        currentPosition <= questionList!!.size -> {
+                            putQuestion()
+                        }
+                        else -> {
+                           val intent = Intent(this, ResultActivity::class.java)
```

Comment:
I would suggest to move this code, which is in else statement to seperate method. You can call this metod 
`private fun goToResultActivity()` and call that method here. That will give for your code more readability.